src/app/test/matchers/toHaveComputedStyle.ts

```diff
@@ -0,0 +1,76 @@
+import CustomMatcherFactories = jasmine.CustomMatcherFactories;
+import CustomMatcher = jasmine.CustomMatcher;
+import CustomMatcherResult = jasmine.CustomMatcherResult;
+
+/**
+ * Custom matchers use a combination of the following resources:
+ * https://stackoverflow.com/questions/42956195/create-custom-jasmine-matcher-using-typescript
+ * https://jasmine.github.io/tutorials/custom_matcher
+ *
+ * Remember to update matcher-types.d.ts when making changes
+ */
+
+type StyleValue = string | number;
+
+function validateStyles(
+  actual: HTMLElement,
+  expected: Partial<CSSStyleDeclaration>,
+  callback: (
+    key: string,
+    actualStyle: StyleValue,
+    expectedStyle: StyleValue
+  ) => CustomMatcherResult
+): CustomMatcherResult {
+  const computedStyle = getComputedStyle(actual);
```

Comment:
I'm not sure how much and what checking is done by getComputedStyle.

Assuming it is is the browser default; then we probably want to make some assertions about the `actual` target here, like not null, is a DOM Element, is attached, etc..